src/app/auth/services/auth.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { BehaviorSubject, EMPTY, finalize, Observable, switchMap, tap } from 'rxjs';
+import { SessionStorageService } from './session-storage.service';
+import { User } from 'src/app/app-model';
+import { InitialUserAdminData, PostResponse } from './auth-model';
+import { environment } from 'src/environments/environment';
+import { UserStoreService } from 'src/app/user/services/user-store.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  private isAuthorized$$ = new BehaviorSubject<boolean>(false);
+  readonly isAuthorized$: Observable<boolean> = this.isAuthorized$$.asObservable();
+
+  isRegistered: boolean = false;
+
+  authLoader$ = new BehaviorSubject<boolean>(false);
+
+  // InitialUserAdminData login as admin. It allows to avoid 403 error
+  user: User = InitialUserAdminData;
+
+  constructor(private _http: HttpClient, private _storage: SessionStorageService, private _userStoreService: UserStoreService) {}
+
+  getUser(): User {
+    return this.user;
+  }
+
+  login(user: User): Observable<User> {
+    this.authLoader$.next(true);
+    return this._http.post(`${environment.baseUrl}/login`, user).pipe(
+      finalize(() => {
+        this.authLoader$.next(false);
+      }),
+      switchMap((response: any) => {
+        if (response.successful === true && response.result !== undefined) {
+          const token = response.result;
+          this.isAuthorized$$.next(true);
+          this._storage.setToken(token);
+          this.user.accessToken = token;
+          return this._userStoreService.getUser();
+        }
+        return EMPTY;
+      })
+    );
+  }
+
+  register(user: User): Observable<PostResponse> {
+    this.authLoader$.next(true);
+    return this._http.post<PostResponse>(`${environment.baseUrl}/register`, user).pipe(
+      finalize(() => {
+        this.authLoader$.next(false);
+      }),
+      tap(response => {
+        if (response.successful === true) {
+          this.user = { ...user };
+        }
+      })
+    );
+  }
+
+  logout(): Observable<any> {
```

Comment:
Try to avoid `any`, use interface from back end part.
```
login(body: UserModel): Observable<SuccessfulRequest<string> | FailedRequest>;
register(body: UserModel): Observable<SuccessfulRequest<string> | FailedRequest>;
logout(token: string): Observable<void | FailedRequest>;
```